src/interface/src/app/map/map.component.ts

```diff
@@ -84,16 +98,35 @@ export class MapComponent implements AfterViewInit {
       }),
       onEachFeature: (feature, layer) => layer.bindPopup(this.popupService.makeDetailsPopup(feature.properties.shape_name))
     });
-    this.map.addLayer(boundaryLayer);
-    layers.addOverlay(boundaryLayer, "HUC-12");
+    this.map.addLayer(this.HUC12BoundariesLayer);
   }
 
-  ngAfterViewInit(): void {
-    var control_layers = this.initMap();
-    this.initCalMapperLayer(control_layers);
-    this.boundaryService.getBoundaryShapes().subscribe((boundary) => {
-      this.boundary = boundary;
-      this.initBoundaryLayer(control_layers);
-    });
+  // Toggle which base layer is shown.
+  changeBaseLayer() {
+    if (this.baseLayerType == BaseLayerType.Terrain) {
```

Comment:
Use === rather than == for equality checks. == can cause error prone type coercions.